### DISCUSSION TITLE
Trying to Run different airfoil (Naca0018) Instead of the airfoilopt(Naca0012)

### DISCUSSION THREAD

**Discussion by vaths1** [mdolab/MACH-Aero#145] [Q&A] [answered]
Hello,
I see the coordinates used in the tutorial ([MACH-Aero](https://github.com/mdolab/MACH-Aero/tree/main)/[tutorial](https://github.com/mdolab/MACH-Aero/tree/main/tutorial)/[airfoilopt](https://github.com/mdolab/MACH-Aero/tree/main/tutorial/airfoilopt)/[mesh](https://github.com/mdolab/MACH-Aero/tree/main/tutorial/airfoilopt/mesh)
/n0012.dat])  are in a continuous format with them starting from the leading edge and going through the trailing edge and at the last the leading edge coordinates are repeated again once again. 

Because the the coordinates I am trying to work with are direcrtly from the uiuc, seling format which seem to be giving me an error and impossible to mesh. 
the error being, 

![image](https://github.com/mdolab/MACH-Aero/assets/151889110/156a36f9-8243-40ee-8d44-33b529bdb36f)

and when i try to visualize 

![image](https://github.com/mdolab/MACH-Aero/assets/151889110/85dedd2c-6a2e-4210-ada6-aea8bf166c63)

I would like to know or understand how this can be done for any other airfoil or any other closed geometry curves?

https://github.com/mdolab/MACH-Aero/blob/main/tutorial/airfoilopt/mesh/n0012.dat

I would really appreciate asistance with this issue!

**Comment by marcomangano** [comment, marked as answer]
Hello, the issue here seems related to either a spurious point, an open curve, or both. You should convert the original coordinates to the format you correctly identified in the `.dat` file, otherwise pyHyp will fail. Try [preFoil](https://github.com/mdolab/prefoil/tree/main) to fix the airfoil coordinates (can that handle UIUC formatting @akleb ?). It includes a NACA airfoil generator as well. Otherwise, if preFoil does not fit your needs and you only have to work on a few airfoils,  you can manually (or write a short script to)  edit the .dat files.

**Reply by vaths1** [reply]
Hello, 
Yes, you are right, it was spurious indeed,  exactly because it was an open curve. The coordinates from the UIUC formatting had an open TE. I had tried with airfoiltools site and was able to generate the coordinates with a closed TE and was able to mesh well as anticipated without the need of the prefoil as well. Thank you so much for responding.
